OpenSim/Examples/ExampleHopperDevice/helperMethods.h

```diff
@@ -184,7 +195,7 @@ inline void simulate(Model& model, SimTK::State& state, bool saveStatesFile)
 
     // Simulate until the user presses ESC (or enters 'q' if visualization has
     // been disabled).
-    while (true) {
+    while (!simulateOnce) {
```

Comment:
Perhaps I am missing something, but this will result in no simulation, not even once. For it to pass through once the condition would have to start true and become false after the first pass.